h/session.py

```diff
@@ -96,6 +98,7 @@ def _user_preferences(user):
 
 def includeme(config):  # pragma: no cover
     settings = config.registry.settings
+    secure = urlparse(settings.get("h.app_url")).scheme == "https"
```

Comment:
This code assumes that a given instance of h is always accessed over the same protocol and this will match the `APP_URL` env var that we use in various contexts when we don't have a request handy.